presto-main/src/main/java/com/facebook/presto/cost/CostCalculatorUsingExchanges.java

```diff
@@ -108,13 +116,29 @@ public PlanNodeCostEstimate visitOutput(OutputNode node, Void context)
         @Override
         public PlanNodeCostEstimate visitFilter(FilterNode node, Void context)
         {
-            return cpuCost(getStats(node.getSource()).getOutputSizeInBytes());
+            // not accessed columns does not cost
+            Set<Symbol> usedSymbols = DependencyExtractor.extractUnique(node.getPredicate());
+            Set<Symbol> notUsedSymbols = Sets.difference(new HashSet<>(node.getSource().getOutputSymbols()), usedSymbols);
+
+            PlanNodeStatsEstimate filterStats = getStats(node.getSource());
+            PlanNodeStatsEstimate filteredFilterStats = PlanNodeStatsEstimate.buildFrom(filterStats).removeSymbolStatistics(notUsedSymbols).build();
```

Comment:
consider `s/filteredFilterStats/onlyUsedSymbolsStats/`

Why unused columns do not cost? Are we always producing dictionary blocks in filter operator?